android/src/main/java/com/onegini/mobile/sdk/reactnative/RNOneginiSdk.kt

```diff
@@ -209,58 +231,67 @@ class RNOneginiSdk(reactContext: ReactApplicationContext) : ReactContextBaseJava
-                }
+        when (url) {
+            null -> promise.rejectWithNullError("url", "String")
+            else -> {
+                oneginiSDK.oneginiClient.userClient.getAppToWebSingleSignOn(
+                    Uri.parse(url),
+                    object : OneginiAppToWebSingleSignOnHandler {
+                        override fun onSuccess(oneginiAppToWebSingleSignOn: OneginiAppToWebSingleSignOn) {
+                            promise.resolve(OneginiAppToWebSingleSignOnMapper.toWritableMap(oneginiAppToWebSingleSignOn))
+                        }
 
-                override fun onError(error: OneginiAppToWebSingleSignOnError) {
-                    promise.reject(error.errorType.toString(), error.message)
-                }
+                        override fun onError(error: OneginiAppToWebSingleSignOnError) {
+                            promise.reject(error.errorType.toString(), error.message)
+                        }
+                    }
+                )
             }
-        )
+        }
+
     }
 
     @ReactMethod
     fun submitCustomRegistrationAction(customAction: String?, identityProviderId: String?, token: String?, promise: Promise) {
-        identityProviderId ?: promise.rejectWithNullError("identityProviderId", "String").run { return }
-
-        val action = registrationManager.getSimpleCustomRegistrationAction(identityProviderId)
-
-        if (action == null) {
-            return promise.reject(OneginiWrapperErrors.IDENTITY_PROVIDER_NOT_FOUND.code, OneginiWrapperErrors.IDENTITY_PROVIDER_NOT_FOUND.message)
-        }
+        when (identityProviderId) {
+            null -> promise.rejectWithNullError("identityProviderId", "String")
+            else -> {
+                val action = registrationManager.getSimpleCustomRegistrationAction(identityProviderId)
 
-        when (customAction) {
-            Constants.CUSTOM_REGISTRATION_ACTION_PROVIDE -> {
-                try {
-                    action.returnSuccess(token)
-                    promise.resolve(null)
-                } catch (exception: OneginiReactNativeException) {
-                    promise.reject(OneginiWrapperErrors.REGISTRATION_NOT_IN_PROGRESS.code, OneginiWrapperErrors.REGISTRATION_NOT_IN_PROGRESS.message)
+                if (action == null) {
+                    return promise.reject(OneginiWrapperErrors.IDENTITY_PROVIDER_NOT_FOUND.code, OneginiWrapperErrors.IDENTITY_PROVIDER_NOT_FOUND.message)
                 }
-            }
-            Constants.CUSTOM_REGISTRATION_ACTION_CANCEL -> {
-                try {
-                    action.returnError(Exception(token))
-                    promise.resolve(null)
-                } catch (exception: OneginiReactNativeException) {
-                    promise.reject(OneginiWrapperErrors.REGISTRATION_NOT_IN_PROGRESS.code, OneginiWrapperErrors.REGISTRATION_NOT_IN_PROGRESS.message)
+
+                when (customAction) {
```

Comment:
WDYT about extracting the code into smaller separate methods? 

```
 val action = registrationManager.getSimpleCustomRegistrationAction(identityProviderId)

                if (action == null) {
                    return promise.reject(OneginiWrapperErrors.IDENTITY_PROVIDER_NOT_FOUND.code, OneginiWrapperErrors.IDENTITY_PROVIDER_NOT_FOUND.message)
                }
```
and 

```
 when (customAction) {
                    Constants.CUSTOM_REGISTRATION_ACTION_PROVIDE -> {
                        try {
                            action.returnSuccess(token)
                            promise.resolve(null)
                        } catch (exception: OneginiReactNativeException) {
                            promise.reject(OneginiWrapperErrors.REGISTRATION_NOT_IN_PROGRESS.code, OneginiWrapperErrors.REGISTRATION_NOT_IN_PROGRESS.message)
                        }
                    }
                    Constants.CUSTOM_REGISTRATION_ACTION_CANCEL -> {
                        try {
                            action.returnError(Exception(token))
                            promise.resolve(null)
                        } catch (exception: OneginiReactNativeException) {
                            promise.reject(OneginiWrapperErrors.REGISTRATION_NOT_IN_PROGRESS.code, OneginiWrapperErrors.REGISTRATION_NOT_IN_PROGRESS.message)
                        }
                    }
                    else -> {
                        promise.reject(OneginiWrapperErrors.PARAMETERS_NOT_CORRECT.code, OneginiWrapperErrors.PARAMETERS_NOT_CORRECT.message + ". Incorrect customAction supplied: $customAction")
                    }
                }
```

Reply:
There is already a story to move this into a useCase, and also a story to competely rewrite this functionality. I think we can split it then if still needed.